### PR TITLE
Filter menu

### PR DESCRIPTION
**Pull request by HelenHMaher**
- automatically selects 'all Time' and 'all Genres'
- when 'all Time' and 'all Genres' are selected by the user all other tiles are deselected
- when 'all Time' and 'all Genres' are selected and the filters are submitted, all times and all genres are displayed for the user
- if zero time filters and/or zero genre filters are selected, 'all Time' and/or 'all Genres' are automatically selected for the user
- if every time filter is saved as selected in the DB, 'all Time' will NOT be automatically selected
- if every genre filter is saved as selected in the DB, 'all Genres' WILL be automatically selected

### CODE REVIEW

File: src/NOTES.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// -- Movie "skip" feature
+// -- "You liked the same movie as a friend!" Notification UI
+// -- Search Friends: improved UI
+// -- enhanced filter selection Logic
```
Comment (fanckush) [inline src/NOTES.js]: do you still need this file? if not you can delete it

**Review by fanckush — DISMISSED** [review]

**Comment by HelenHMaher** [comment]
Okay, there are still some bugs going on but this pull request felt like it was getting out of hand so I decided to get it to a point where it was not actively breaking anything and then start another branch that chases down some of the bugs.  I think this might involved refactoring some of the components I wrote at the beginning.

this pull requests:

1) changes the filter component to allow the user to toggle "all time" and "all genres"

2) changes the match notification to reflect the UI we have for it in figma (I needed to create some new icons for this)

3) makes some minor changes to the MatchChard that allows for empty state handling so that the page won't continuously try to reload whenever the "toSwipe" array is empty (in some cases there might be no more movies in our database to display)

File: src/components/MatchPage/MatchCard/MatchCard.js [hunk 2]
```diff
@@ -42,16 +41,18 @@ export const MatchCard = ({ decision, reset }) => {
     ));
 
   useEffect(() => {
-    if (isMounted.current && preLoadArray.length < 5) {
+    console.log('useEffect1');
```
Comment (fanckush) [inline src/components/MatchPage/MatchCard/MatchCard.js]: you forgot to remove this

File: src/components/MatchPage/MatchCard/MatchCard.js [hunk 3]
```diff
@@ -42,16 +41,18 @@ export const MatchCard = ({ decision, reset }) => {
...
     } else {
-      isMounted.current = true;
+      setIsMounted(true);
     }
   }, [preLoadArray]);
 
   //when a decision is made it triggers an axios call
 
   useEffect(() => {
+    console.log('useEffect2');
```
Comment (fanckush) [inline src/components/MatchPage/MatchCard/MatchCard.js]: same here

**Comment by fanckush** [comment]
left some comments but it's looking good (code wise)

File: src/components/MatchPage/MatchCard/MatchCard.js (hunk 3 above)
Comment (HelenHMaher) [reply src/components/MatchPage/MatchCard/MatchCard.js]: thanks!  good catch :)

**Review by fanckush — APPROVED** [review]

**Pull request merged**